lib/task-queue/Storage.js

```diff
@@ -37,29 +39,35 @@ const CREATE_TASKS_TABLE = `CREATE TABLE moz_tasks (
 // SQL for droping the moz_tasks table
 const DROP_TASKS_TABLE = "DROP TABLE moz_tasks;";
 
+let INSTANCE = null;
+
 class Storage {
   /**
-   * Handles database storage needs for the task queue
+   * Handles database and filesystem storage needs for the task queue
    */
   constructor() {
```

Comment:
it would be cleaner to have an `instance` method.

Should the instance not exist, the constructor would be called and object returned, otherwise the instance is returned.

```javascript

let INSTANCE = null;

class Storage {
  constructor() {
  },

  instance() {
    if (!INSTANCE) {
      INSTANCE = new Storage();
    }
    return INSTANCE;
 }
}
```